lib/src/model/inheriting_container.dart

```diff
@@ -0,0 +1,617 @@
+  }
+
+  @override
+  Iterable<Field> get declaredFields => allFields.where((f) => !f.isInherited);
+
+  /// Add a single Field to _fields.
+  ///
+  /// If [f] is not specified, pick the FieldElement from the PropertyAccessorElement
+  /// whose enclosing class inherits from the other (defaulting to the getter)
+  /// and construct a Field using that.
+  void _addSingleField(
+      PropertyAccessorElement getterElement,
+      PropertyAccessorElement setterElement,
+      Set<PropertyAccessorElement> inheritedAccessors,
+      [FieldElement f]) {
+    var getter =
+        ContainerAccessor.from(getterElement, inheritedAccessors, this);
+    var setter =
+        ContainerAccessor.from(setterElement, inheritedAccessors, this);
+    // Rebind getterElement/setterElement as ModelElement.from can resolve
+    // MultiplyInheritedExecutableElements or resolve Members.
+    getterElement = getter?.element;
+    setterElement = setter?.element;
+    assert(!(getter == null && setter == null));
+    if (f == null) {
+      // Pick an appropriate FieldElement to represent this element.
+      // Only hard when dealing with a synthetic Field.
+      if (getter != null && setter == null) {
+        f = getterElement.variable;
+      } else if (getter == null && setter != null) {
+        f = setterElement.variable;
+      } else {
+        /* getter != null && setter != null */
+        // In cases where a Field is composed of two Accessors defined in
+        // different places in the inheritance chain, there are two FieldElements
+        // for this single Field we're trying to compose.  Pick the one closest
+        // to this class on the inheritance chain.
+        if (setter.enclosingElement is Class &&
+            (setter.enclosingElement as Class)
+                ._isInheritingFrom(getter.enclosingElement)) {
+          f = setterElement.variable;
+        } else {
+          f = getterElement.variable;
+        }
+      }
+    }
+    Field field;
+    if ((getter == null || getter.isInherited) &&
+        (setter == null || setter.isInherited)) {
+      // Field is 100% inherited.
+      field = ModelElement.fromPropertyInducingElement(f, library, packageGraph,
+          enclosingContainer: this, getter: getter, setter: setter);
+    } else {
+      // Field is <100% inherited (could be half-inherited).
+      // TODO(jcollins-g): Navigation is probably still confusing for
+      // half-inherited fields when traversing the inheritance tree.  Make
+      // this better, somehow.
+      field = ModelElement.fromPropertyInducingElement(f, library, packageGraph,
+          getter: getter, setter: setter);
+    }
+    _allFields.add(field);
+  }
+
+  Iterable<Method> _declaredMethods;
+
+  @override
+  Iterable<Method> get declaredMethods =>
+      _declaredMethods ??= element.methods.map((e) {
+        return ModelElement.from(e, library, packageGraph) as Method;
+      });
+
+  List<TypeParameter> _typeParameters;
+
+  // a stronger hash?
```

Comment:
Is this a TODO or something?

Reply:
this looks like leftovers from previous cut and paste refactoring adventures.  removing